src/pytorch_lightning/loggers/csv_logs.py

```diff
@@ -195,7 +195,7 @@ def log_hyperparams(self, params: Union[Dict[str, Any], Namespace]) -> None:
         self.experiment.log_hparams(params)
 
     @rank_zero_only
-    def log_metrics(self, metrics: Dict[str, float], step: Optional[int] = None) -> None:
+    def log_metrics(self, metrics: Dict[str, Union[Tensor, float]], step: Optional[int] = None) -> None:
```

Comment:
updated to use `metrics: Dict[str, Union[Tensor, float]]` across all loggers for `metrics`.